### PR TITLE
Add R-MAT model

### PR DESCRIPTION
**Pull request by panpiort8**
solves #17

### CODE REVIEW

**Comment by kudkudak** [comment]
Mergujemy? :)

**Comment by Mazzza** [comment]
Dzisiaj to ogarnę. Dzięki za przypomnienie :)

W dniu wt., 2.11.2021 o 09:15 Stanislaw Jastrzebski <
***@***.***> napisał(a):

**Comment by panpiort8** [comment]
@Mazzza maybe today is a good day?

File: src/huggingmolecules/configuration/configuration_rmat.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+RMAT_CONFIG_ARCH = {
+    'rmat_4M': 'https://drive.google.com/uc?id=12J6yE8W92mrdnzpKemdmPDWO52EQf_J6'
+}
+
+@dataclass
+class RMatConfig(PretrainedConfigMixin):
+    d_atom: int = 37
+    d_edge: int = 46
+    d_model: int = 768
+    init_type: str = 'normal'
```
Comment (Mazzza) [inline src/huggingmolecules/configuration/configuration_rmat.py:15]: there was uniform in original code, but i think its marginal thing

File: src/huggingmolecules/configuration/configuration_rmat.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    encoder_n_layers: int = 10
+    encoder_n_attn_heads: int = 12
+
+    ffn_activation: str = 'LeakyReLU'
+    ffn_n_layers: int = 2
+    ffn_d_hidden: int = 1536
+    ffn_d_output: int = 768
+
+    generator_aggregation: str = 'mean'
```
Comment (Mazzza) [inline src/huggingmolecules/configuration/configuration_rmat.py:30]: we used attention pooling, not mean

File: src/huggingmolecules/configuration/configuration_rmat.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    generator_n_layers: int = 1
+    generator_d_outputs: int = 1
+    generator_features_generators: List[str] = None
```
Comment (Mazzza) [inline src/huggingmolecules/configuration/configuration_rmat.py:33]: i think, that we could also do another config for rmat with rdkit features, as there are slightly different hps for generator

**Review by Mazzza — CHANGES_REQUESTED** [review]

File: src/huggingmolecules/configuration/configuration_rmat.py (hunk 1 above)
Comment (panpiort8) [reply src/huggingmolecules/configuration/configuration_rmat.py]: done

File: src/huggingmolecules/configuration/configuration_rmat.py (hunk 2 above)
Comment (panpiort8) [reply src/huggingmolecules/configuration/configuration_rmat.py]: done

File: src/huggingmolecules/configuration/configuration_rmat.py (hunk 3 above)
Comment (panpiort8) [reply src/huggingmolecules/configuration/configuration_rmat.py:34]: done

**Pull request merged**